openedx/openedx-platform
# refactor: update codejail test condition

## Description
Update test condition to skip test for codejail if the codejailservice api is not enabled. 
This condition would help tutor to be able to run the test against its `codejailservice`


## Supporting information
This PR fixes https://github.com/openedx/wg-developer-experience/issues/156


## Testing instructions
Before this change, running the below in tutor
```
tutor dev run lms bash
  lms> unset DJANGO_SETTINGS_MODULE
  lms> unset SERVICE_VARIANT
  lms> export EDXAPP_TEST_MONGO_HOST=mongodb
  lms> pytest 'xmodule/capa/safe_exec/tests/test_safe_exec.py::TestSafeOrNot::test_cant_do_something_forbidden'
```
would result in the test being skipped.
After this change, we should be able to run the test, it should pass and not be skipped.
Please note that it can only be run in tutor if the setting `ENABLE_CODEJAIL_REST_SERVICE` is `True` and `CODE_JAIL_REST_SERVICE_HOST` are adequately set.
You many also need to use a custom docker-compose file should the one tutor create in its env, not have `lms` depend on `codejailservice` as this service must also be running when you run `tutor dev run lms bash`


Fixes: https://github.com/openedx/wg-developer-experience/issues/156
Private-ref: https://tasks.opencraft.com/browse/BB-7705
